core/src/main/java/com/sequenceiq/cloudbreak/service/filesystem/FileSystemConfigService.java

```diff
@@ -0,0 +1,88 @@
+package com.sequenceiq.cloudbreak.service.filesystem;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUser;
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUserRole;
+import com.sequenceiq.cloudbreak.domain.FileSystem;
+import com.sequenceiq.cloudbreak.repository.FileSystemRepository;
+import com.sequenceiq.cloudbreak.service.AuthorizationService;
+
+@Service
+public class FileSystemConfigService {
+
+    private static final String ACCESS_DENIED_MESSAGE = "Access denied!";
+
+    @Inject
+    private FileSystemRepository fileSystemRepository;
+
+    @Inject
+    private AuthorizationService authService;
+
+    public Set<FileSystem> retrievePrivateFileSystems(@Nonnull IdentityUser user) {
+        return fileSystemRepository.findForUser(user.getUserId());
+    }
+
+    public FileSystem create(@Nonnull IdentityUser user, @Nonnull FileSystem fileSystem) {
+        setUserDataRelatedFields(user, fileSystem);
+        authService.hasWritePermission(fileSystem);
+        return fileSystemRepository.save(fileSystem);
+    }
+
+    public FileSystem getPrivateFileSystem(String name, @Nonnull IdentityUser user) {
+        FileSystem fileSystem = fileSystemRepository.findByNameInUser(name, user.getUserId());
+        authService.hasReadPermission(fileSystem);
+        return fileSystem;
+    }
+
+    public FileSystem get(Long id) {
+        FileSystem fileSystem = Optional.ofNullable(fileSystemRepository.findOne(id)).orElseThrow(() -> new AccessDeniedException(ACCESS_DENIED_MESSAGE));
```

Comment:
as we earlier discussed we should handle with the same event when a resource is not exists and/or the user has not enough right, so that's why I used AccessDeniedException

Reply:
Response status and exception is the log are two different things